In [ ]:
# Fatemeh Makhloughi
# Machine Learning
# HW5- Logistic Regression for SUV Dataset (Classification problem)
import numpy as np
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import classification_report, confusion_matrix
from sklearn.preprocessing import LabelEncoder
import matplotlib.pyplot as plt

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
df= pd.read_csv('/content/drive/MyDrive/Machine Learning/SUV.csv')

In [ ]:
df

User ID  Gender  Age  EstimatedSalary  Purchased
0    15624510    Male   19            19000          0
1    15810944    Male   35            20000          0
2    15668575  Female   26            43000          0
3    15603246  Female   27            57000          0
4    15804002    Male   19            76000          0
..        ...     ...  ...              ...        ...
395  15691863  Female   46            41000          1
396  15706071    Male   51            23000          1
397  15654296  Female   50            20000          1
398  15755018    Male   36            33000          0
399  15594041  Female   49            36000          1

[400 rows x 5 columns]

In [ ]:
X = df.iloc[:,1:-1].values
Y = df['Purchased'].values

In [ ]:
label_encoder_x= LabelEncoder()
X[:, 0]= label_encoder_x.fit_transform(X[:, 0])

In [ ]:
x_train, x_test, y_train, y_test = train_test_split(X, Y, test_size=0.2, random_state=42,  stratify=Y)

In [ ]:
Stand = StandardScaler()

x_train = Stand.fit_transform(x_train)
x_test = Stand.transform(x_test)

In [ ]:
log_reg = LogisticRegression()
log_reg.fit(x_train, y_train)

LogisticRegression()

In [ ]:
print(log_reg.score(x_test, y_test))

0.8


In [ ]:
y_pred = log_reg.predict(x_test)
print(confusion_matrix(y_test, y_pred))

[[47  4]
 [12 17]]


In [ ]:
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.80      0.92      0.85        51
           1       0.81      0.59      0.68        29

    accuracy                           0.80        80
   macro avg       0.80      0.75      0.77        80
weighted avg       0.80      0.80      0.79        80



This model is correctly identifying 92% of the class 0s, but only 59% of the class 1s.

This model achieved an accuracy of 80%, which is not very good.